libs/zglfw/src/zglfw.zig

```diff
@@ -233,6 +244,147 @@ pub const CursorMode = enum(i32) {
 };
 //--------------------------------------------------------------------------------------------------
 //
+// Joystick
+//
+//--------------------------------------------------------------------------------------------------
+
+pub const Joystick = struct {
+    pub const ButtonAction = enum(u8) {
+        release = 0,
+        press = 1,
+    };
+
+    jid: u4,
+
+    pub const maximum_supported = 16;
+
+    pub fn init(jid: u4) ?Joystick {
+        const self: Joystick = .{ .jid = jid };
+        if (!self.present()) {
+            return null;
+        }
+        return self;
+    }
+
+    pub fn present(self: Joystick) bool {
```

Comment:
Let's call it `isPresent()`.

Reply:
Actually this should be a global function called `isJoystickPresent(jid: u4) bool`. It should not be possible to create an instance of `Joystick` if it is not present on the system.